### DISCUSSION TITLE
Electron density on grid points by FCI

### DISCUSSION THREAD

**Discussion by maryam-khd** [pyscf/pyscf#2687] [Q&A] [answered]
I need to calculate the electron density on grid points after solving the FCI for a simple H2 system. Based on the issue #1334, I have tried to make the density matrix on AO basis although I face with an error since the FCI make_rdm1 function does not have ao_repr attribute and computes the density in MO basis: 
 mol = gto.Mole()
mol.atom = [['H', (0, -1.5/2, 0)], ['H', (0, 1.5/2, 0)]]
mol.unit = 'B'
mol.basis = '6-31G'
mol.spin = 0
mol.charge = 0
mol.build()
grids = dft.gen_grid.Grids(mol)
grids.level = gengrid_level
grids.build()
npoints = grids.coords.shape[0]
e1coords = grids.coords[:npoints//2,:]
e2coords = grids.coords[npoints//2:,:]

# Run Hartree-Fock calculation
mf = mol.RHF().run()
    
fcisolver = fci.FCI(mol, mf.mo_coeff)
e1, ci1 = fcisolver.kernel()

#Calculate Rho on grids
dm1s = fcisolver.make_rdm1(ao_repr=True)
dm1 = dm1s[0] + dm1s[1]
ao_value =  dft.numint.eval_ao(mol, e1coords, deriv=0)
rho = dft.numint.eval_rho(mol, ao_value, dm1)      
       

Moreover I couldn't find how the https://github.com/pyscf/pyscf/blob/master/examples/fci/14-density_matrix.py
can be applied to my case, as I am running spin-restricted case and it is not clear to me which of these matrices is the one I have to use.

**Comment by sunqm** [comment, marked as answer]
As a temporary solution, do not call fcisolver.make_rdm1 with `ao_repr=True`. Then transform the FCI density matrix with the MO coefficients `mf.mo_coeff`
```
dm1 = mf.mo_coeff.dot(fcisolver.make_rdm1()).dot(mf.mo_coeff.T)
```

**Comment by stefanvuckovic1** [comment]
> As a temporary solution, do not call fcisolver.make_rdm1 with `ao_repr=True`. Then transform the FCI density matrix with the MO coefficients `mf.mo_coeff`
> 
> ```
> dm1 = mf.mo_coeff.dot(fcisolver.make_rdm1()).dot(mf.mo_coeff.T)
> ```

What would be an analogous way to compute dm2 for the same case in AO basis?

**Comment by sunqm** [comment]
`einsum('ijkl,pi,qj,rk,sl->pqrs', fcisolver.make_rdm2(), mo_coeff, mo_coeff, mo_coeff, mo_coeff)`
